src/tools/clippy/tests/ui/new_ret_no_self.rs

```diff
@@ -73,7 +73,7 @@ struct T;
 impl T {
     // should not trigger lint
     pub fn new() -> Self {
-        unimplemented!();
+        unimplemented!()
```

Comment:
Wait, why does this have to change? 😅

Reply:
Having to change this feels inconsistent with what's in https://github.com/rust-lang/rust/pull/123590#issuecomment-2089207009, but I'm possibly missing something?

Reply:
it doesn't have to change. I just changed it partly because of preference, partly because this might break on the future. 

(note that comment you linked does accept breakage like this; only saying that changing `return` style is too much)

Reply:
Please change this in a different PR then, I'd rather not entangle unnecessary changes into this one

Reply:
I reverted these changes, so the commit/this PR only include the necessary changes to make this lint not fire for the clippy test